test/common_cuda.py

```diff
@@ -3,12 +3,32 @@
 import torch
 import torch.cuda
 
+class common_cuda(object):
+    __instance = None
 
-TEST_CUDA = torch.cuda.is_available()
-TEST_MULTIGPU = TEST_CUDA and torch.cuda.device_count() >= 2
-CUDA_DEVICE = TEST_CUDA and torch.device("cuda:0")
-TEST_CUDNN = TEST_CUDA and torch.backends.cudnn.is_acceptable(torch.tensor(1., device=CUDA_DEVICE))
-TEST_CUDNN_VERSION = TEST_CUDNN and torch.backends.cudnn.version()
+    def __new__(cls):
+        if cls.__instance == None:
+            cls.__instance = object.__new__(cls)
+        return cls.__instance
+
+    def __init__(self):
```

Comment:
This seems like an awfully roundabout way to ensure that initialization only happens once. A more ordinary thing is to have the object initialize all the attributes, and just make sure you only initialize the object itself once.